packages/stellar-axelar-std-derive/src/contractimpl.rs

```diff
@@ -2,16 +2,25 @@ use itertools::Itertools;
 use proc_macro2::Ident;
 use quote::quote;
 use syn::{
-    parse_quote, AngleBracketedGenericArguments, GenericArgument, ImplItem, ImplItemFn, ItemImpl,
-    PathArguments, PathSegment, ReturnType, Stmt, Type, TypePath, Visibility,
+    parse_quote, AngleBracketedGenericArguments, FnArg, GenericArgument, ImplItem, ImplItemFn,
+    ItemImpl, Pat, PathArguments, PathSegment, ReturnType, Stmt, Type, TypePath, Visibility,
 };
 
 use crate::utils::{parse_env_identifier, PrependStatement};
 
 pub fn contractimpl(impl_block: &mut ItemImpl) -> Result<proc_macro2::TokenStream, syn::Error> {
-    // this needs to be defined before the iteration, because during it, we can't get a reference to the impl block
+    let all_contract_endpoints = all_contract_endpoints(impl_block);
     let any_stateful_endpoints = any_stateful_endpoints(impl_block);
 
+    impl_block
+        .items
+        .iter_mut()
+        .filter_map(all_contract_endpoints)
+        .try_for_each(|method| {
+            instance_ttl_extension(method)?;
+            Ok::<_, syn::Error>(())
+        })?;
```

Comment:
with the rest of my suggestions it's possible to add this to the other iteration, but you can also leave it separate if you prefer that

Reply:
If you want to keep the two iterations separate I suggest you swap them so the storage top up will be the first line in each function